GatherBuddy/FishTimer/FishTimerWindow.cs

```diff
@@ -127,35 +130,65 @@ private void DrawTextHeader(string bait, string spot, int milliseconds)
             case 0: return;
             case -1:
                 var text = "Elapsed Time";
-                ImGui.SameLine(_windowSize.X - ImGui.CalcTextSize(text).X - offset);
+                ImGui.SameLine(_windowSize.X - ImGui.CalcTextSize(text).X - TextMargin);
                 ImGui.Text(text);
                 return;
             default:
                 var secondText = (_milliseconds / 1000.0).ToString("00.0");
-                ImGui.SameLine(_windowSize.X - ImGui.CalcTextSize(secondText).X - offset);
+                ImGui.SameLine(_windowSize.X - ImGui.CalcTextSize(secondText).X - TextMargin);
                 ImGui.Text(secondText);
                 return;
         }
     }
 
-    private void SetSpot(FishingSpot? spot)
+    private string EllipsifyString(string text, float maxWidth)
+    {
+        if (ImGui.CalcTextSize(text).X < maxWidth)
+        {
+            return text;
+        }
+
+        while (text.Length > 0)

```

Comment:
I really do not like this at all, otherwise I'd have done it that way already. But I don't know how to do it less terribly either.

Reply:
What's your concern here?

My thought is that this only happens when the spot changes and spot names are not that long.  I considered doing a binary search, but it felt like more complication than it was worth for an infrequent event that's n=50 at most.

Reply:
I just don't like it. I think it's as efficient as it can reasonably be atm (ImGui is working on some kind of support for automatic ellipses afaik) and doesn't matter either way. :D

Reply:
Do you plan on doing further work on stuff? Because I already PR'd an update to the plugin repo after the first two PRs, I'd wait if you keep on working.

Reply:
I think the only other thing on my mind in the short term is to add an uptime timer for spectral currents when ocean fishing.  That's the only thing I really feel like I'm missing when fishing.

(Longer-term, personally #61 would be really nice (although who knows how to visualize that) or to handle chum better, but I don't think I have time to do either of those soon.)

Reply:
I did some slight changes to this PR but there's an issue that the first line throw after loading the plugin yields no spot name due to _windowSize.X being set only in Draw (i.e. after UpdateSpotText). 
I can't rework it right now but will do so later, unless you do it in the meantime.

Reply:
Oh, good catch.  Sorry, I must have missed that ordering issue.  I am not sure I have a clean idea on how to fix it.  If you do, I'm happy to implement it, otherwise I'll think on it more, or maybe you'll fix it first.

Reply:
Hmm maybe #143 is an option?